### ISSUE TITLE
redis.clients.jedis.exceptions.JedisClusterException: CLUSTERDOWN The cluster is down 

### ISSUE THREAD

**Issue by deepshiv126** [xetorthio/jedis#2144]
We are seeing **CLUSTERDOWN** issue like thousands of times, as a result, we are losing data obtained from Redis Server. 

We observed this is not happening to any particular command, we have observed it occurs to String-based, List based commands. 

Any help is really appreciated.

Redis Library: 3.1.0
Spring Data Redis: 2.2.3.RELEASE
**Redis Server: 5.0.4**

```

org.springframework.dao.InvalidDataAccessApiUsageException: CLUSTERDOWN The cluster is down; 
nested exception is redis.clients.jedis.exceptions.JedisClusterException: CLUSTERDOWN The cluster is down 
 	at org.springframework.data.redis.connection.jedis.JedisExceptionConverter.convert(JedisExceptionConverter.java:69) 
 	at org.springframework.data.redis.connection.jedis.JedisExceptionConverter.convert(JedisExceptionConverter.java:42) 
 	at org.springframework.data.redis.PassThroughExceptionTranslationStrategy.translate(PassThroughExceptionTranslationStrategy.java:44) 
 	at org.springframework.data.redis.FallbackExceptionTranslationStrategy.translate(FallbackExceptionTranslationStrategy.java:42) 
 	at org.springframework.data.redis.connection.jedis.JedisClusterConnection.convertJedisAccessException(JedisClusterConnection.java:777) 
 	at org.springframework.data.redis.connection.jedis.JedisClusterStringCommands.convertJedisAccessException(JedisClusterStringCommands.java:525) 
 	at org.springframework.data.redis.connection.jedis.JedisClusterStringCommands.set(JedisClusterStringCommands.java:139) 
 	at org.springframework.data.redis.connection.DefaultedRedisConnection.set(DefaultedRedisConnection.java:288) 
 	at org.springframework.data.redis.cache.DefaultRedisCacheWriter.lambda$put$0(DefaultRedisCacheWriter.java:90) 
 	at org.springframework.data.redis.cache.DefaultRedisCacheWriter.execute(DefaultRedisCacheWriter.java:242) 
 	at org.springframework.data.redis.cache.DefaultRedisCacheWriter.put(DefaultRedisCacheWriter.java:87) 
 	at org.springframework.data.redis.cache.RedisCache.put(RedisCache.java:150) 
 	at org.springframework.cache.transaction.TransactionAwareCacheDecorator.put(TransactionAwareCacheDecorator.java:105) 
 	at org.springframework.cache.interceptor.AbstractCacheInvoker.doPut(AbstractCacheInvoker.java:87) 
 	at org.springframework.cache.interceptor.CacheAspectSupport$CachePutRequest.apply(CacheAspectSupport.java:820) 
 	at org.springframework.cache.interceptor.CacheAspectSupport.execute(CacheAspectSupport.java:429) 
 	at org.springframework.cache.interceptor.CacheAspectSupport.execute(CacheAspectSupport.java:345) 
 	at org.springframework.cache.interceptor.CacheInterceptor.invoke(CacheInterceptor.java:61) 
 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) 
 	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:93) 
 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) 
 	at org.springframework.aop.interceptor.AbstractTraceInterceptor.invoke(AbstractTraceInterceptor.java:134) 
 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) 
 	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:689) 

```

**Comment by sazzad16** [comment]
I can see that you found and #1461 first. Please read and understand everything in that issue, especially following two comments:
- https://github.com/xetorthio/jedis/issues/1461#issuecomment-344038224 
- https://github.com/xetorthio/jedis/issues/1461#issuecomment-346553862 

The gist is:
- Jedis does not throw that exception on its own. It is Redis that send that error response. So you should concern about Redis (cluster).
- Users experienced this when Redis was busy in other works.

**Comment by deepshiv126** [comment]
Thanks for the response @sazzad16. 
Yes! I found #1461 issue and read each comment, but their observations were not at all relatable to our situation; Also their mentioned, Redis command was not used in our scenarios.

And I have to agree to your points - and I too noticed in the [code](https://github.com/xetorthio/jedis/blob/master/src/main/java/redis/clients/jedis/Protocol.java#L125), Jedis is just delegating the error message which it received from Redis Server and this is happening only when Redis is very busy.

**Comment by deepshiv126** [comment]
Update: This is an issue with AWS Elasticache Redis and nothing to do with Jedis Library or open-source Redis.
Now, this issue is fixed on AWS Elasticache Redis too.

**Comment by maybhi** [comment]
> Update: This is an issue with AWS Elasticache Redis and nothing to do with Jedis Library or open-source Redis.
> Now, this issue is fixed on AWS Elasticache Redis too.

@deepshiv126 in which version of ElastiCache Redis Cluster(Cluster Mode Enabled) it's resolved?

**Comment by deepshiv126** [comment]
@maybhi - The issue is resolved on **5.0.6** version and aws not yet released 6.#.# version(hopefully they will release it sooner).
